zmon-security-common/src/main/java/org/zalando/zmon/security/grafanatoken/GrafanaTokenService.java

```diff
@@ -0,0 +1,75 @@
+package org.zalando.zmon.security.grafanatoken;
+
+import com.google.common.collect.Lists;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.MACVerifier;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+import javaslang.control.Try;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.RememberMeAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.zalando.zmon.security.authority.ZMONRoleToAuthority;
+import org.zalando.zmon.security.authority.ZMonRole;
+import org.zalando.zmon.security.jwt.JWTServiceProperties;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Collections;
+
+public class GrafanaTokenService {
```

Comment:
Why do you decided to extract this class? Why this logic cannot be a part of `GrafanaRememberMeServices`?

Reply:
following the same pattern as [JWTService.java](https://github.com/zalando-zmon/zmon-controller/blob/master/zmon-security-common/src/main/java/org/zalando/zmon/security/jwt/JWTService.java) and [TvTokenService.java](https://github.com/zalando-zmon/zmon-controller/blob/master/zmon-security-common/src/main/java/org/zalando/zmon/security/tvtoken/TvTokenService.java) where token service class manages token handling, whereas `ZMonRememberMeServices` subclasses implement the interface and delegate logic to token service.